backend/services/stripe_service.py

```
@@ -0,0 +1,270 @@
+                return False, None, 0, "No user email found"
+
+            if credits <= 0:
+                logger.error(f"Invalid credits in checkout session {session.get('id')}")
+                return False, user_email, 0, "Invalid credit amount"
+
+            logger.info(
+                f"Checkout completed: {user_email} purchased {credits} credits "
+                f"(package: {package_id}, session: {session.get('id')})"
+            )
+
+            return True, user_email, credits, f"Successfully purchased {credits} credits"
```

Ruff (0.14.10)

Consider moving this statement to an `else` block

(TRY300)